README.md

```diff
@@ -6,46 +6,108 @@
 
 # FlutterBleLib
 
-A library for all your Bluetooth Low Energy needs in Flutter. Internally utilises Polidea's 
-[MultiPlatformBleAdapter](https://github.com/Polidea/MultiPlatformBleAdapter), 
+A library for all your Bluetooth Low Energy needs in Flutter. Internally utilises Polidea's
+[MultiPlatformBleAdapter](https://github.com/Polidea/MultiPlatformBleAdapter),
 which runs on [RxAndroidBle](https://github.com/Polidea/RxAndroidBle)
- and [RxBluetoothKit](https://github.com/Polidea/RxBluetoothKit).
+and [RxBluetoothKit](https://github.com/Polidea/RxBluetoothKit).
 
 ## BLE Simulator
-This library supports [BLEmulator](https://github.com/Polidea/blemulator_flutter), the BLE simulator. 
+
+This library supports [BLEmulator](https://github.com/Polidea/blemulator_flutter), the BLE simulator.
 The simulation allows one to develop without physical smartphone or BLE peripheral and use one's production BLE–related code in automated testing.
 
+## Installation
+
+To use this plugin, add `flutter_ble_lib` as a [dependency in your pubspec.yaml file](https://pub.dev/packages/flutter_ble_lib/versions/2.1.0).
+
+### Android
+
+Set `minSDKVersion` in `[project]/android/app/build.gradle` file to 18.
+
+```gradle
+defaultConfig {
+  ...
+  // Set min SDK version to 18
+  minSdkVersion 18
+  targetSdkVersion 28
+  ...
+}
+```
+
+Add `tools:replace="android:label"` rule to `<application>` element in main `AndroidManifest.xml` file.
+
+```xml
+<application
+  android:name="io.flutter.app.FlutterApplication"
+  android:label="your_app"
+  android:icon="@mipmap/ic_launcher"
+  tools:replace="android:label">
+  ...
+</application>
+```
+
+Add `BLUETOOTH` and `ACCESS_FINE_LOCATION` permissions in main `AndroidManifest.xml` file.
+
+```xml
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    package="com.example.your_app">
+
+  <uses-permission android:name="android.permission.BLUETOOTH"/>
+  <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
+
+  ...
+</manifest>
+```
```

Comment:
Permissions are mentioned in Usage section and it bloats the README. If you think it's necessary, perhaps it should be in a separate file, only linked here?

Reply:
Oh, yes, there is a notice about `ACCESS_FINE_LOCATION`. Then I would rather remove it from Usage section and leave it more detailed in Installation. 

For me, from a user perspective, it's more convinient to have the whole instruction inline in README instead of linked to separate file. I don't remember if I ever saw such solution on any GitHub page. As a user I always look for installation instructions on top or README and I get very frustrated when I have to search anywhere else.

Reply:
I can create a new patch version `RxAndroidBle 1.7.1` that would have `ACCESS_FINE_LOCATION` and `android:label` removed – Manifest Merger should do the rest. This should make it easier to integrate. WDYT?

Reply:
The less you have to change in native project the better. I think it's a good idea.

Reply:
ACCESS_FINE_LOCATION would still be required for scanning to work, and that's the only way to obtain peripherals for now.

Reply:
But there would be only a need to have this runtime permission. No need to add it in the manifest because a dependency would have it.

Reply:
Ah, nice one. Yeah, seems to be a good idea!

Reply:
Technically I have released `1.7.1` which could be integrated in MBA and here. Should help with both manifest entries